nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -527,8 +527,10 @@ ObstacleLayer::updateBounds(
 
     const sensor_msgs::msg::PointCloud2 & cloud = obs.cloud_;
 
-    double sq_obstacle_max_range = obs.obstacle_max_range_ * obs.obstacle_max_range_;
-    double sq_obstacle_min_range = obs.obstacle_min_range_ * obs.obstacle_min_range_;
+    const unsigned int max_cells = cellDistance(obs.obstacle_max_range_);
+    const unsigned int min_cells = cellDistance(obs.obstacle_min_range_);
+    const uint64_t sq_obstacle_max_range = max_cells * max_cells;
```

Comment:
I'm pretty sure you can use unsigned int, that would be a ~200,000 km distance for the max approx. 4B value at at 5cm resolution. That's plenty large.